### PR TITLE
Remove Onboarding related Properties from LocalData (EXPOSUREAPP-5375)

### PR DESCRIPTION
**Pull request by kolyaopahle**
### Note that this PR does not provide any migration functionality from LocalData -> Settings, this will be done in a later PR

This PR moves all Onboarding related properties of LocalData into OnboardingData which is a injectable Singleton and uses the FlowPreference class to supply the values

### CODE REVIEW

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/storage/LocalData.kt [hunk 1]
```diff
@@ -23,89 +23,6 @@ object LocalData {
...
-     ****************************************************/
-
-    /**
-     * Gets the boolean if the user has completed the onboarding
-     * from the EncryptedSharedPrefs
-     *
-     * @return boolean if user is onboarded
-     */
-    fun isOnboarded(): Boolean = getSharedPreferenceInstance().getBoolean(
-        CoronaWarnApplication.getAppContext().getString(R.string.preference_onboarding_completed),
```
Comment (d4rken) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/storage/LocalData.kt:37]: Remove from strings.xml too?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/storage/OnboardingData.kt [hunk 2]
```diff
@@ -0,0 +1,48 @@
+package de.rki.coronawarnapp.storage
+
+import android.content.Context
+import de.rki.coronawarnapp.util.di.AppContext
+import de.rki.coronawarnapp.util.preferences.clearAndNotify
+import de.rki.coronawarnapp.util.preferences.createFlowPreference
+import org.joda.time.Instant
+import javax.inject.Inject
+
+class OnboardingData @Inject constructor(
```
Comment (d4rken) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/storage/OnboardingData.kt:10]: Where does the migration happen?

Comment (kolyaopahle) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/storage/OnboardingData.kt:10]: Nowhere yet, i still want to align with @jurajkusner on the specifics of how to migrate

Comment (kolyaopahle) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/storage/OnboardingData.kt]: Migration will be done for all preferences in a seperate PR

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/storage/OnboardingSettings.kt [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+    var isBackgroundCheckDone: Boolean
+        get() = prefs.getBoolean(BACKGROUND_CHECK_DONE, false)
+        set(value) = prefs.edit { putBoolean(BACKGROUND_CHECK_DONE, value) }
+
+    fun clear() = prefs.clearAndNotify()
+
+    companion object {
+        private const val ONBOARDING_COMPLETED_TIMESTAMP = "onboarding.done.timestamp"
+        private const val BACKGROUND_CHECK_DONE = "onboarding.background.checked"
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/storage/OnboardingSettings.kt:36]: should not the keys match old keys? or a migration will implemented to copy over these data ?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/onboarding/OnboardingActivity.kt [hunk 4]
```diff
@@ -71,9 +75,12 @@ class OnboardingActivity : AppCompatActivity(), LifecycleObserver, HasAndroidInj
...
 
     fun completeOnboarding() {
-        LocalData.isOnboarded(true)
-        LocalData.onboardingCompletedTimestamp(System.currentTimeMillis())
-        settings.lastChangelogVersion.update { BuildConfigWrap.VERSION_CODE }
+        onboardingSettings.onboardingCompletedTimestamp = timeStamper.nowUTC
+
+        settings.lastChangelogVersion.update {
+            BuildConfigWrap.VERSION_CODE
+        }
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/onboarding/OnboardingActivity.kt:82]: what about `isOnboarded `? it is removed ,but no saved using the new class

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/storage/OnboardingSettings.kt [hunk 5]
```diff
@@ -0,0 +1,38 @@
+package de.rki.coronawarnapp.storage
+
+import android.content.Context
+import androidx.core.content.edit
+import de.rki.coronawarnapp.util.di.AppContext
+import de.rki.coronawarnapp.util.preferences.clearAndNotify
+import org.joda.time.Instant
+import javax.inject.Inject
+
+class OnboardingSettings @Inject constructor(
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/storage/OnboardingSettings.kt:10]: this class is injected in Application ,is not better to annotate it with Singleton?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/onboarding/OnboardingActivity.kt (hunk 4 above)
Comment (Kolpa) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/onboarding/OnboardingActivity.kt:82]: `isOnboarded` has been replaced with a pure getter based on `onboardingCompletedTimestamp != null` to create a unified datasource and remove desync opportunities

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/storage/OnboardingSettings.kt (hunk 5 above)
Comment (Kolpa) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/storage/OnboardingSettings.kt:10]: whoops, missed that, absolutely

**Review by d4rken — APPROVED** [review]
lgtm

**Review by mtwalli — APPROVED** [review]
lgtm

**Pull request merged**